eth/downloader/peer.go

```diff
@@ -57,11 +60,10 @@ type peer struct {
 	blockIdle   int32 // Current block activity state of the peer (idle = 0, active = 1)
 	receiptIdle int32 // Current receipt activity state of the peer (idle = 0, active = 1)
 	stateIdle   int32 // Current node data activity state of the peer (idle = 0, active = 1)
-	rep         int32 // Simple peer reputation
 
-	blockCapacity   int32 // Number of blocks (bodies) allowed to fetch per request
-	receiptCapacity int32 // Number of receipts allowed to fetch per request
-	stateCapacity   int32 // Number of node data pieces allowed to fetch per request
+	blockThroughput   float64 // Number of blocks (bodies) measured to be retrievable per second
```

Comment:
@fjl @karalabe There is a (partly subjective) train of thought among developers that floats should be avoided unless they have to be used or offer a significant benefit: http://www.javapractices.com/topic/TopicAction.do?Id=213
Also https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html

For example, rounding errors accumulating over time can lead to surprises, and logging of the value may show fractions not useful for the given data type (in this case throughput).

In this perhaps floats will not cause any problems - but that's also a sign that they are probably not needed - why cannot the throughput be modelled as a plain uint?

Usually floats are chosen as a lazy solution to something that can be modelled with uints - the question is - do you really need a fractional part? Can the throughput be modelled over a time unit where a fractional part is not needed? E.g. blocks per minute/hour instead of per second.

I'm fine merging the PR with floats here - just wanted to raise this as I think we need to beware of any increase in complexity of the code base - even if the extra complexity is only the use of a float instead of a uint.